src/NodeInfo.cpp

```diff
@@ -43,3 +43,10 @@ void NodeInfo::setNode(size_t nodeId, struct in_addr nodeIP) { //change node IP
 	}
 	it->second = nodeIP;
 }
+
+void NodeInfo::callForEachNode(std::function<void (struct in_addr *)> callback)
+{
+	std::unique_lock<std::mutex> uLock(nodeFilesMtx);
```

Comment:
I believe there's no need to lock nodeFiles mutex when we only read nodeMap

Reply:
And what about when during reading file transfer completes? moreover i think it is ok, because it forces deterministic changes of NodeInfo state

Reply:
ok, so at least change name of mutex